src/driver-roccat-kone-aimo.c

```diff
@@ -0,0 +1,876 @@
+					    ARRAY_LENGTH(report_rates));
+	ratbag_profile_set_report_rate(profile, report_rate);
+
+	ratbag_profile_for_each_resolution(profile, resolution) {
+		dpi_x = setting_report->xres[resolution->index] * 50;
+		dpi_y = setting_report->yres[resolution->index] * 50;
+		//if (!(setting_report->dpi_mask & (1 << resolution->index))) {
+		//	/* the profile is disabled, overwrite it */
+		//	dpi_x = 0;
+		//	dpi_y = 0;
+		//}
+
+		ratbag_resolution_set_resolution(resolution, dpi_x, dpi_y);
+		ratbag_resolution_set_cap(resolution,
+					  RATBAG_RESOLUTION_CAP_SEPARATE_XY_RESOLUTION);
+		resolution->is_active = (resolution->index == setting_report->current_dpi);
+
+		ratbag_resolution_set_dpi_list_from_range(resolution, ROCCAT_MIN_DPI, ROCCAT_MAX_DPI);
+	}
+
+	buf = drv_data->profiles[profile->index];
+	roccat_set_config_profile(device, profile->index, ROCCAT_CONFIG_KEY_MAPPING);
+	rc = ratbag_hidraw_get_feature_report(device, ROCCAT_REPORT_ID_KEY_MAPPING,
+					      buf, ROCCAT_REPORT_SIZE_PROFILE);
+
+	msleep(10);
+
+	if (rc < ROCCAT_REPORT_SIZE_PROFILE)
+		return;
+
+    ratbag_profile_for_each_button(profile, button)
+		roccat_read_button(button);
+
+	if (!roccat_crc_is_valid(device, buf, ROCCAT_REPORT_SIZE_PROFILE))
+		log_error(device->ratbag,
+			  "Error while reading profile %d, continuing...\n",
+			  profile->index);
+
+	log_raw(device->ratbag, "profile: %d %s:%d\n",
+		buf[2],
+		__FILE__, __LINE__);
+}
+
+static int
+roccat_probe(struct ratbag_device *device)
+{
+	int rc;
+	struct ratbag_profile *profile;
+	struct roccat_data *drv_data;
+	int active_idx;
+
+	rc = ratbag_open_hidraw(device);
+	if (rc)
+		return rc;
+
+	if (!ratbag_hidraw_has_report(device, ROCCAT_REPORT_ID_KEY_MAPPING)) {
+		ratbag_close_hidraw(device);
+		return -ENODEV;
+	}
+
+	drv_data = zalloc(sizeof(*drv_data));
+	ratbag_set_drv_data(device, drv_data);
+
+	/* profiles are 0-indexed */
+	ratbag_device_init_profiles(device,
+				    ROCCAT_PROFILE_MAX,
+				    ROCCAT_NUM_DPI,
+				    ROCCAT_BUTTON_MAX,
+				    ROCCAT_LED_MAX);
+
+	ratbag_device_for_each_profile(device, profile)
+		roccat_read_profile(profile);
+
+	active_idx = roccat_current_profile(device);
+	if (active_idx < 0) {
+		log_error(device->ratbag,
+			  "Can't talk to the mouse: '%s' (%d)\n",
+			  strerror(-active_idx),
+			  active_idx);
+		rc = -ENODEV;
+		goto err;
+	}
+
+	list_for_each(profile, &device->profiles, link) {
+		if (profile->index == (unsigned int)active_idx) {
+			profile->is_active = true;
+			break;
+		}
+	}
+
+	log_raw(device->ratbag,
+		"'%s' is in profile %d\n",
+		ratbag_device_get_name(device),
+		profile->index);
+
+	return 0;
+
+err:
+	free(drv_data);
+	ratbag_set_drv_data(device, NULL);
+	return rc;
+}
+
+static void
+roccat_remove(struct ratbag_device *device)
+{
+	ratbag_close_hidraw(device);
+	free(ratbag_get_drv_data(device));
+}
+
+struct ratbag_driver roccat_driver = {
+	.name = "Roccat Kone AIMO",
+	.id = "roccat",
```

Comment:
Set this to `.id = "roccat-kone-aimo",` for now until the Roccat drivers are unified, see also https://github.com/libratbag/libratbag/pull/1001.